### PR TITLE
Model support

### PR DESCRIPTION
**Pull request by mmschlk**
#### Improvements and Ease of Use
- adds a `class_label` parameter to `TabularExplainer` and `Explainer` to specify the class label to be explained for classification models [#271](https://github.com/mmschlk/shapiq/issues/271)
- adds support for `PyTorch` models to `Explainer` [#272](https://github.com/mmschlk/shapiq/issues/272)

#### Bug Fixes
- fixes a bug that `RandomForestClassifier` models were not working with the `TreeExplainer` [#273](https://github.com/mmschlk/shapiq/issues/273)

#### The What I changed:
- main changes are done to explainer.utils where I add a wrapper callable which extracts class_index values from the prediciton functions. 
- fix for RandomForest is simple, moving where the scaling of each tree is applied down after the values are potentially transformed into probabilities
- added a lot of additional tests checking for the efficency property, checking that shapiq can handle different model classes (more soon) and adding comparison of shapiq to shap

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 ## Changelog
 
+### v1.1.1 (2024-11-11)
```
Comment (hbaniecki) [inline CHANGELOG.md:3]: Add date of (just before) the PyPI release.

Comment (mmschlk) [reply CHANGELOG.md:3]: Yes, I will release v1.1.1 today with the hotfixes.

File: shapiq/explainer/utils.py [hunk 2]
```diff
@@ -104,36 +113,59 @@ def get_predict_function_and_model_type(model, model_class):
...
+        predictions = model(data).detach().numpy()
+        return self._get_class_based_predictions(predictions)
 
+    def predict_tensorflow(self, model, data):
+        predictions = model.predict(data, verbose=0)
+        return self._get_class_based_predictions(predictions)
 
-def predict_tf_first(m, d):
-    return m.predict(d, verbose=0)[:, 0]
+    def _get_class_based_predictions(self, predictions):
```
[the remark this answers is not in the document]
Comment (mmschlk) [reply shapiq/explainer/utils.py:163]: Yea I refactored the logic from the class and named it `_predict_function_with_class_index`

File: shapiq/explainer/tabular.py [hunk 3]
```diff
@@ -79,6 +80,7 @@ def __init__(
         self,
         model,
         data: np.ndarray,
+        class_label: Optional[int] = None,
```
[the remark this answers is not in the document]
Comment (mmschlk) [reply shapiq/explainer/tabular.py:83]: Yes. Good point. Changed to class_index.

**Review by hbaniecki — APPROVED** [review]
